### PR TITLE
Anyclip Bid Adapter : refactor bid adapter

### PR DESCRIPTION
**Pull request by Chucky-choo**
## Type of change
- [x] Updated bidder adapter

### CODE REVIEW

File: modules/anyclipBidAdapter.js [hunk 1]
```diff
@@ -1,213 +1,222 @@
...
-   * @param {*} bidderRequest
-   * @return {ServerRequest}
-   */
-  buildRequests: (validBidRequests, bidderRequest) => {
-    const bidRequest = validBidRequests[0];
-
-    let refererInfo;
-    if (bidderRequest && bidderRequest.refererInfo) {
-      refererInfo = bidderRequest.refererInfo;
+    if (config.getConfig('coppa')) {
```
Comment (ChrisHuie) [inline modules/anyclipBidAdapter.js:101]: Can you please access coppa using `bidderRequest.ortb2.regs.coppa` instead of through config.

Comment (Chucky-choo) [reply modules/anyclipBidAdapter.js]: Hello @ChrisHuie,
Thank you for the review. We have decided to remove this block, as we have the full ORTB2 object in the request. We will proceed with this value on the adserver.
Let me know if you need any further adjustments!

**Comment by patmmccann** [comment]
@Chucky-choo please note the comment above about adding linter warnings, probably jsdoc syntax errors

**Comment by Chucky-choo** [comment]
Hello @patmmccann. Thank you for the review. I solved all the warnings related to jsdoc. Please let me know if everything is fine now

File: modules/anyclipBidAdapter.js [hunk 2]
```diff
@@ -1,213 +1,219 @@
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
 import {registerBidder} from '../src/adapters/bidderFactory.js';
-import {BANNER} from '../src/mediaTypes.js';
-import {deepAccess, isArray, isFn, logError, logInfo} from '../src/utils.js';
-import {config} from '../src/config.js';
+import {parseSizesInput, isFn, deepAccess, getBidIdParameter, logError, isArray} from '../src/utils.js';
+import {getAdUnitSizes} from '../libraries/sizeUtils/sizeUtils.js';
+import {findIndex} from '../src/polyfill.js';
```
Comment (patmmccann) [inline modules/anyclipBidAdapter.js:5]: Do you still need this?

**Comment by patmmccann** [comment]
It appears you have some massive code duplication with several other adapters. Are these companies affiliated with you? Can you move them into commonly imported functions?

**Review by ncolletti — APPROVED** [review]

**Review by patmmccann — APPROVED** [review]

**Pull request merged**